src/converter.cpp

```diff
@@ -96,30 +97,37 @@ bool Converter::transcode(char *src, char *dst)
     ret = avformat_write_header(encoder->fmtCtx, NULL);
     if (ret < 0)
     {
-        fprintf(stderr, "Error occurred when opening output file: %s\n", av_err2str(ret));
+    //    fprintf(stderr, "Error occurred when opening output file: %s\n", av_err2str(ret));
         flag = false;
         goto end;
     }
 
+    // av_dump_format(encoder->fmtCtx, 0, encoder->filename, 1);
+
     //read video data from multimedia files to write into destination file
     while(av_read_frame(decoder->fmtCtx, decoder->pkt) >= 0)
     {
+        stream_index = decoder->pkt->stream_index;
```

Comment:
you set this variable just for temporarily store stream_index and used by two av_packet_rescale_ts in one loop?

maybe you can just use decoder-pkt->stream_index or you can declare the variable within loop scope

Reply:
I have tried your suggestion, but it is not feasible, because to set audio and video synchronization in transcoder.cpp function, it need to use the corresponding encoder->videoCodecCtx or encoder->audioCodecCtx to implement.

Reply:
so what does it have to do with stream_index?